Yoast/wordpress-seo
# Correctly implemented count_user_posts call in class-frontend.php

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where the "Show archives for authors without posts in search results?" option does not work.

## Relevant technical choices:

* Used the same logic as in https://github.com/Yoast/wordpress-seo/pull/13569/files#diff-48d16ddb9f75f32b6ef92d466733b7cbR76-R83.

## Test instructions

This PR can be tested by following these steps:

Set default options:
* Set `SEO` > `Search Appearance` > `Archives` > `Author archives settings` > `Author archives` to `Enabled`.
* Set `SEO` > `Search Appearance` > `Archives` > `Author archives settings` > `Show author archives in search results?` to `Yes`.

* Create a new author.
* Set `SEO` > `Search Appearance` > `Archives` > `Author archives settings` > `Show archives for authors without posts in search results?` to `No`.
* Go to the archive for the new author, make sure its robots meta value is set to `noindex,follow`.
* Set `SEO` > `Search Appearance` > `Archives` > `Author archives settings` > `Show archives for authors without posts in search results?` to `Yes`.
* Go to the archive for the new author make sure the robots meta value is set to `max-snippet:-1, max-image-preview:large, max-video-preview:-1` (because of [this PR](https://github.com/Yoast/wordpress-seo/pull/13541)).

## UI changes

## Documentation

## Quality assurance


Fixes #12567
Fixes #10369
